projects/core/src/i18n/i18next/i18next-plugins/i18next-logger-plugin.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+
+import { LoggerModule } from 'i18next';
+import { LoggerService } from '../../../logger';
+
+//TODO: add unit tests if approved
+@Injectable({
+  providedIn: 'root',
+})
+export class I18nextLoggerPlugin {
+  logger = inject(LoggerService);
+
+  getPlugin(): LoggerModule {
+    return {
+      type: 'logger',
+      log: (args) => this.logger.log(...args),
+      warn: (args) => this.logger.warn(...args),
+      error: (args) => this.logger.error(...args),
```

Comment:
[MEDIUM]

Don't we need to deconstruct args with spread `...`?
```suggestion
      log: (...args) => this.logger.log(...args),
      warn: (...args) => this.logger.warn(...args),
      error: (...args) => this.logger.error(...args),
```

Reply:
Good point. In fact, it doesn't change the way how args are logged, but it's good to keep consistency.

Reply:
Just for reference:
we continued this discussion in a new PR: https://github.com/SAP/spartacus/pull/17512#discussion_r1227791790